var/spack/repos/builtin/packages/ucc/package.py

```diff
@@ -0,0 +1,46 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Ucc(AutotoolsPackage, CudaPackage):
+    """UCC is a collective communication operations API and library that is
+    flexible, complete, and feature-rich for current and emerging programming
+    models and runtimes."""
+
+    homepage = "https://openucx.github.io/ucc/"
+    url = "https://github.com/openucx/ucc/archive/refs/tags/v1.1.0.tar.gz"
+
+    version("1.2.0", sha256="c1552797600835c0cf401b82dc89c4d27d5717f4fb805d41daca8e19f65e509d")
+    version("1.1.0", sha256="74c8ba75037b5bd88cb703e8c8ae55639af3fecfd4428912a433c010c97b4df7")
+    version("1.0.0", sha256="d3b4aa7004bf339d35952a1699a6e408064ba578bdc93861f5f07527ad0a5e8c")
+
+    variant("cuda", default=False, description="Enable CUDA TL", when="@1.1:")
+    variant("nccl", default=False, description="Enable NCCL TL")
+    variant("rccl", default=False, description="Enable RCCL TL", when="@1.1:")
+    variant("ucp", default=True, description="Enable UCP TL")
+
+    conflicts("cuda@12:", when="@1.1", msg="UCC 1.1 supports CUDA <12")
+    conflicts("~cuda", when="+nccl", msg="UCC NCCL TL requires CUDA")
+
+    depends_on("autoconf", type="build")
+    depends_on("automake", type="build")
+    depends_on("libtool", type="build")
+
+    depends_on("cuda", when="+nccl")
```

Comment:
Wondering if we should avoid this directive. If we have `+nccl~cuda` we will lose all the constraints from `CudaPackage` but still depends on `cuda` in this package. This is not problematic in practice, as long as there is at least another `CudaPackage` with `+cuda` in the runtime part of the DAG - but seems wrong modeling.

Since `nccl` depends on `cuda` itself, is this necessary?

Reply:
UCC's build looks for CUDA when building with NCCL support so it requires both.
`+nccl ~cuda` is listed as a conflict so it shouldn't be a problem.

Reply:
Yeah, what you don't get are the constraints from the `CUDA` package which are active only when `+cuda`

Reply:
I could dump the old versions and make `+nccl` a conditional variant on `+cuda`, would that work?

Reply:
Sure, that works. Let me know if it gets too complicated though.

Reply:
@alalazo fixed, I hope.